### ISSUE TITLE
Disable tracing to the httpCall() API

### ISSUE THREAD

**Issue by gongcon** [envoyproxy/envoy#14785]
Since 1.15.0, tracing is enabled for `httpCall()` in lua filter. Is there an option to disable it? We have a lua filter calling a sidecar service. It creates too many spans than our limited budget.

**Comment by asraa** [comment]
cc @dio

**Comment by gongcon** [comment]
any update?

**Comment by dio** [comment]
Sorry for the late reply. So the current API is:

``` lua
local headers, body = handle:httpCall(cluster, headers, body, timeout, asynchronous)
```

Since the tracing option is only enabled for "asynchronous" calls, I think we can allow this last arg as a table as well:

```lua
handle:httpCall(cluster, headers, body, timeout, true)
-- By default httpCall is async.
handle:httpCall(cluster, headers, body, timeout, {["disableTracing"]: true}) 
```

**Comment by gongcon** [comment]
Thanks @dio for the response. There's another problem of it. It does not honor the tracing config in both listener and route. The http call creates spans even we set overall_sampling to 0, which disabled tracing of requests coming into envoy.

**Comment by wbpcode** [comment]
closed by #22871

**Comment by jdomag** [comment]
@dio 
Can you explain a bit why below approach doesn't have an effect and we need to disable tracing directly in httCall()?
My LUA filter takes the request and  forward it to service providing authorization based on the headers of request. 
I use envoy sampling so every 10% of total traffic gets sampled. I would like to see in the full span (request lifecycle) the requests to the authorization service as well and it works as expected. However requests that were not sampled and got x_b3_sampled header set to 0 are sometimes sampled in LUA filter. I figured I can copy x_b3_sampled header from original request and add this header to a httpCall(), but it doesn't work - the request is sampled anyway. Why is that? 
Second question, with the approach introduced in: https://github.com/envoyproxy/envoy/pull/22871  is it possible to achieve that 10% of traffic that got sampled presents on spans calls made by LUA filter? Or if I disable tracing via options, no traffic will be sampled and I won't be able to see delays or time that request spend on LUA filter calls?
